snapcraft/_store.py

```diff
@@ -847,7 +848,20 @@ def collaborate(snap_name, key):
 
     assertion = _sign_assertion(snap_name, assertion, key, 'developers')
 
-    store.push_assertion(snap_id, assertion, 'developers')
+    try:
+        store.push_assertion(snap_id, assertion, 'developers')
+    except storeapi.errors.StoreValidationError as e:
+        if e.error_list[0]['code'] == 'revoked-uploads':
+            print("This will revoke the following uploads: {}".format(
+                e.error_list[0]['extra']))
+            if input("Are you sure you want to continue (y/N): ") == "y":
+                store.push_assertion(
+                    snap_id, assertion, 'developers', force=True)
+            else:
+                print("The collaborators for this snap haven't been altered")
+                sys.exit(1)
```

Comment:
Is it common to exit with error if the user cancels the action? It doesn't sound ok to me, but I don't know.
Anyway, the right way to exit is raising an exception with the message, that then will be printed by main and exit.